[Bug]: After sending the first transaction RAM is filled up

### Version

`next`: `7fe907609cc0bb791cb97aa3db499f0c7a33d622`

### Other packages versions

rust 1.85

### What happened?

The node seems to start up properly, but after sending 1 transaction (I was using the faucet, in particular, but looks like it is generated by transaction) starts to consume a lot of memory until it runs out. At some point it start increasing by 2GB per second.

<img width="956" alt="Image" src="https://github.com/user-attachments/assets/22ebee55-4506-452f-bbb3-2265c5adf2cb" />

I paste an error that I got in the output a couple fields down.

### What should have happened?

The node should not run out of memory (Macbook M3 Max with 36 GB RAM).

### How can this be reproduced?

```bash
mkdir data accounts
miden-node store dump-genesis > genesis.toml
miden-node bundled bootstrap --data-directory data --accounts-directory accounts
miden-node bundled start --rpc.url http://0.0.0.0:57291 --data-directory data
miden-faucet init
miden-faucet create-faucet-account -t POL -d 10 -m 1000000
miden-faucet start --config miden-faucet.toml
```
Then create a mint through the faucet and it should start happening after a couple seconds.

### Relevant log output

```shell
2025-03-19T18:32:59.728929Z  INFO write_block_to_db: miden-store: crates/store/src/db/mod.rs:483: new
2025-03-19T18:32:59.729036Z  INFO write_block_to_db: miden-store: crates/store/src/db/mod.rs:483: close, time.busy: 97.3µs, time.idle: 10.2µs
2025-03-19T18:32:59.729092Z ERROR store.rpc/ApplyBlock:store.server.apply_block:apply_block: miden-store: crates/store/src/state.rs:236: error: block applying was cancelled because of closed channel on database side rpc.service: "store.rpc", rpc.method: "ApplyBlock"
2025-03-19T18:32:59.729117Z  INFO store.rpc/ApplyBlock:store.server.apply_block:apply_block: miden-store: crates/store/src/state.rs:236: close, time.busy: 1.21ms, time.idle: 358µs rpc.service: "store.rpc", rpc.method: "ApplyBlock"
2025-03-19T18:32:59.729099Z ERROR apply_block: miden-store: crates/store/src/db/mod.rs:470: error: SQLite error
2025-03-19T18:32:59.729129Z ERROR store.rpc/ApplyBlock:store.server.apply_block: miden-store: crates/store/src/server/api.rs:292: error: status: Internal, message: "block applying was cancelled because of closed channel on database side", details: [], metadata: MetadataMap { headers: {} } rpc.service: "store.rpc", rpc.method: "ApplyBlock"
2025-03-19T18:32:59.729134Z  INFO apply_block: miden-store: crates/store/src/db/mod.rs:470: close, time.busy: 47.4µs, time.idle: 193µs
2025-03-19T18:32:59.729142Z  INFO store.rpc/ApplyBlock:store.server.apply_block: miden-store: crates/store/src/server/api.rs:292: close, time.busy: 1.72ms, time.idle: 373µs rpc.service: "store.rpc", rpc.method: "ApplyBlock"
2025-03-19T18:32:59.729203Z ERROR store.rpc/ApplyBlock: tower_http::trace::on_failure: /Users/santiagopittella/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tower-http-0.6.2/src/trace/on_failure.rs:93: response failed, classification: Code: 13, latency: 2 ms rpc.service: "store.rpc", rpc.method: "ApplyBlock"
2025-03-19T18:32:59.729231Z  INFO store.rpc/ApplyBlock: miden_node_utils::tracing::grpc: crates/utils/src/tracing/grpc.rs:36: close, time.busy: 1.86ms, time.idle: 396µs rpc.service: "store.rpc", rpc.method: "ApplyBlock"
2025-03-19T18:32:59.729431Z ERROR block_builder.build_block:block_builder.commit_block:store.client.apply_block: miden-block-producer: crates/block-producer/src/store/mod.rs:232: error: gRPC client error
2025-03-19T18:32:59.729449Z  INFO block_builder.build_block:block_builder.commit_block:store.client.apply_block: miden-block-producer: crates/block-producer/src/store/mod.rs:232: close, time.busy: 122µs, time.idle: 2.81ms
2025-03-19T18:32:59.729459Z ERROR block_builder.build_block:block_builder.commit_block: miden-block-producer: crates/block-producer/src/block_builder/mod.rs:210: error: failed to apply block to store
```

--- a/crates/store/src/db/sql/mod.rs
+++ b/crates/store/src/db/sql/mod.rs
@@ -1223,9 +1223,10 @@ pub fn apply_block(
     accounts: &[BlockAccountUpdate],
 ) -> Result<usize> {
     let mut count = 0;
+    // Note: ordering here is important as the relevant tables have FK dependencies.
     count += insert_block_header(transaction, block_header)?;
-    count += insert_notes(transaction, notes)?;
     count += upsert_accounts(transaction, accounts, block_header.block_num())?;
+    count += insert_notes(transaction, notes)?;
     count += insert_transactions(transaction, block_header.block_num(), accounts)?;
     count += insert_nullifiers_for_block(transaction, nullifiers, block_header.block_num())?;
     Ok(count)

